src/lib/PostgresMetaFunctions.ts

```diff
@@ -57,4 +63,101 @@ export default class PostgresMetaFunctions {
       return { data: null, error: { message: 'Invalid parameters on function retrieve' } }
     }
   }
+
+  async create({
+    name,
+    schema = 'public',
+    params,
+    definition,
+    rettype = 'void',
+    language = 'sql',
+  }: {
+    name: string
+    schema?: string
+    params?: string[]
+    definition: string
+    rettype?: string
+    language?: string
+  }): Promise<PostgresMetaResult<PostgresFunction>> {
+    const sql = `
+      CREATE FUNCTION ${ident(schema)}.${ident(name)}
+      ${params && params.length ? `(${params.join(',')})` : '()'}
+      RETURNS ${rettype || 'void'}
+      AS '${definition}'
+      LANGUAGE ${language}
+      RETURNS NULL ON NULL INPUT;
+      `
+    const { error } = await this.query(sql)
+    if (error) {
+      return { data: null, error }
+    }
+    return await this.retrieve({ name, schema })
+  }
+
+  async update(
```

Comment:
@i-pip updating function definition with `CREATE OR REPLACE FUNCTION` might get tricky (i.e. function dropped right before `CREATE OR REPLACE FUNCTION` transaction would create an entirely new function) so feel free to skip for now.

Reply:
@w3b6x9 I added this to `async create(` instead, this way it's still possible to update the function definition without creating an entirely new function while trying to update with `PATCH`. Problem though is that I think this breaks RESTful conventions because `CREATE OR REPLACE` on `async create(` would be performing an update with `POST`.